daemon/cmd/local_node_sync.go

```diff
@@ -120,16 +124,45 @@ func (ini *localNodeSynchronizer) getK8sLocalNode(ctx context.Context) (*slim_co
 	return nil, ctx.Err()
 }
 
+func (ini *localNodeSynchronizer) getK8sLocalCiliumNode(ctx context.Context) (*v2.CiliumNode, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	for ev := range ini.K8sCiliumLocalNode.Events(ctx) {
+		ev.Done(nil)
+		if ev.Kind == resource.Upsert {
+			return ev.Object, nil
+		}
```

Comment:
The wait period in case the CiliumNode does not exist could be reduced by returning immediately once the Sync event is received. At that point, we are sure that the CiliumNode will never be observed, as it could only be created by the agent itself at a later stage. That should also remove the need for the additional timeout configured for the context. Another possibility could be replacing this whole logic with a one-shot get, but that would be slightly worse in terms of performance (assuming that the CiliumLocalNode is also used elsewhere, as the informer would then be shared).

Reply:
Yeah good call, I reworked this to return immediately in case of sync, otherwise wait for an upsert.